bats/sanity-check/20-selftests.bats

```diff
@@ -1667,10 +1669,23 @@ selftest_cpuset_unsorted() {
     local -a cpuset=($($SANDSTONE --dump-cpu-info | sort -rnk2,3 |
                            awk '/^[0-9]/ { printf "%d,", $1; }'))
     cpuset=${cpuset%,}          # remove last comma
+    local -a cpuinfo=(`$SANDSTONE --dump-cpu-info | awk '/^[0-9]/ { print $1; }'`)
 
     selftest_cpuset_unsorted "$cpuset" "${cpuinfo[@]}"
```

Comment:
This should cause the test function to be called like
`selftest_cpuset_unsorted "3,2,1,0" 0 1 2 3`

So it shouldn't work the way you modified the function, with:
```sh
    local cpuinfo=$1
```
That would only see one of the values.

So I'm surprised the tests passed

Reply:
Yes when i tested locally this worked. I even tried this and worked 
**local cpuinfo=("$@")**
but since $1 worked i kept that. 
Another thing i tried was remove shift and use $2 that seemed to work too . but i just left that alone

Reply:
It works because it's incomplete (my fault, the existing test failed to do that). The test is actually missing a check that there were no more CPUs.

Here's the fix:
```patch
diff --git i/bats/sanity-check/20-selftests.bats w/bats/sanity-check/20-selftests.bats
index 81f3b718..dcfb4d2b 100644
--- i/bats/sanity-check/20-selftests.bats
+++ w/bats/sanity-check/20-selftests.bats
@@ -29,6 +29,7 @@ test_yaml_absent() {
     printf "Key is not absent:\\n"
     printf "query:      %s\n" "$query"
     printf "value =     %s\n" "${yamldump[$query]}"
+    return 1
 }
 
 test_yaml_numeric() {
@@ -1650,11 +1651,11 @@ selftest_cpuset_unsorted() {
 
     # Get all the processor numbers
     i=0
-    local cpuinfo=$1
-    for expected_logical in "${cpuinfo[@]}"; do
+    for expected_logical; do
         test_yaml_numeric "/cpu-info/$i/logical" "value == $expected_logical"
         i=$((i + 1))
     done
+    test_yaml_absent "/cpu-info/$i/logical"
 }
 
 @test "cpuset=number (inverse order)" {
```

Please apply, reorder the commits for clean-up and I'll approve.